### PR TITLE
fix argocd login error message from install-argo-cd.sh

### PR DESCRIPTION
**Pull request by alsuren**
This is a bit of a hack.

The tradeoff is that it will break passwords that are stored in people's browsers, but it will make `argocd login` work if you switch to another person's cluster and back, and it will remove a warning.

It basically does what Jan suggests in docs/maintenance.md, but it does it automatically:

> Please note: We currently use the default Argo CD admin password, which is the argocd-server pod name and generated on first start. After upgrading Argo CD the pod name will change. If you want to change the password to change the new pod name, you can remove the admin.password and admin.passwordMtime keys in the argocd-secret and restart the server pod.

There is a better solution, which is to pin down the password in a sealed secret. I haven't looked into that though.

### CODE REVIEW

**Review by frigus02 — COMMENTED** [review]
This seems fine for dev clusters.

But since the script can also be executed for production, I would feel safer if we didn't just reset the password automatically without asking. It's not a user visible change, so it's probably fine. Just seems, hm, a bit unsafe to me.

File: infrastructure/scripts/install-argo-cd.sh [hunk 1]
```diff
@@ -62,6 +62,11 @@ kustomize build ../kubernetes/argocd/apps/$ENVIRONMENT |
 export ARGOCD_OPTS='--port-forward --port-forward-namespace argocd'
 
 POD_NAME=$(kubectl get pods -n argocd | grep --only-matching 'argocd-server-[^ ]*')
-argocd login --username admin --password $POD_NAME
+if ! argocd login --username admin --password $POD_NAME; then
+	echo "looks like argocd has been upgraded. Resetting the password."
+	KUBE_EDITOR='sed -i "" "/admin.password/d"' kubectl edit secret -n argocd argocd-secret -o json
+	kubectl rollout restart -n argocd deployment argocd-server
+	POD_NAME=$(kubectl get pods -n argocd | grep --only-matching 'argocd-server-[^ ]*')
```
[the remark this answers is not in the document]
Comment (alsuren) [reply infrastructure/scripts/install-argo-cd.sh:69]: ah yes.

I guess I could add a RESET_PASSWORD environment variable to make this optional.

I am also in danger of over-polishing this turd, if we would actually rather use sealed secrets instead.

[the remark this answers is not in the document]
Comment (frigus02) [reply infrastructure/scripts/install-argo-cd.sh:69]: Yeah, I do wonder if we should rather create a sealed secret for that 😄

[the remark this answers is not in the document]
Comment (alsuren) [reply infrastructure/scripts/install-argo-cd.sh]: I haven't wrapped my head around sealed secrets enough. Happy to close this PR if it moves us in the wrong direction.

**Review by frigus02 — APPROVED** [review]

**Pull request merged**